src/apps/competitions/models.py

```diff
@@ -23,6 +28,107 @@ def __str__(self):
+
+        logger.info("Checking for needed migrations on competition pk=%s, current phase: %s, next phase: %s" %
+                    (self.pk, current_phase.index, next_phase.index))
+
+        # Check for next phase and see if it has auto_migration enabled
+        if next_phase and next_phase.auto_migration:
+            if next_phase.start < now() and current_phase.end < now():
+                self.apply_phase_migration(current_phase, next_phase)
+
+    def apply_phase_migration(self, current_phase, next_phase):
+        '''
+        Does the actual migrating of submissions from current_phase to next_phase
+
+        :param current_phase: The phase object to transfer submissions from
+        :param next_phase: The new phase object we are entering
+        '''
+        logger.info("Checking for submissions that may still be running competition pk=%s" % self.pk)
+
+        if current_phase.submissions.filter(status='Scoring').exists():
+            logger.info('Some submissions still marked as processing for competition pk=%s' % self.pk)
+            self.is_migrating_delayed = True
+            self.save()
+            return
+        else:
+            logger.info("No submissions running for competition pk=%s" % self.pk)
+
+        logger.info('Doing phase migration on competition pk=%s from phase: %s to phase: %s' %
+                    (self.pk, current_phase.index, next_phase.index))
+
+        if self.is_migrating:
+            logger.info('Trying to migrate competition pk=%s, but it is already being migrated!' % self.pk)
+            return
+
+        self.is_migrating = True
+        self.save()
+
+        try:
+            submission_list = []
+            submissions = Submission.objects.filter(phase=current_phase)
+
+            for submission in submissions:
+                submission_list.append(submission)
+
+            participants = {}
+
+            for s in submission_list:
+                if s.is_migrated is False:
+                    participants[s.participant] = s
+
+            for participant, submission in participants.items():
+                logger.info('Moving submission %s over' % submission)
+
+                file_args = {
+                    "result": submission.result,
+                    "owner": submission.owner,
+                    "data": submission.data,
+                }
+
+                new_submission = Submission(
+                    participant=participant,
+                    phase=next_phase,
+                    **file_args
+                )
+                new_submission.save(ignore_submission_limit=True)
+                new_submission.start()
+
+                submission.is_migrated = True
+                submission.save()
+
+        except Submission.DoesNotExist:
+            pass
+
+        # To check for submissions being migrated, does not allow to enter new submission
+        next_phase.is_migrated = True
+        next_phase.save()
+
+        # TODO: ONLY IF SUCCESSFUL
```

Comment:
What about this todo?

Reply:
We could add a field to phases that is something like `watch_list` or something like that which saves the pks of the submissions it created, then in the celery job that looks for phases that need to be migrated, also run
```python
Phase.objects.annotate(running_subs=Count(Subquery(Submission.objects.filter(pk__in=OuterRef('watch_list')).exclude(
    status__in=['Finished', 'Failed', 'Cancelled']).values('pk')))).filter(running_subs=0, is_migrating=True).update(
    is_migrating=False)
```
Pretty sure this will update phases to `is_migrate=False` if their `watch_list` submissions are in a "done" status. Should do it all in one query too ;) 

@ckcollab Thoughts on this as a solution to this todo?